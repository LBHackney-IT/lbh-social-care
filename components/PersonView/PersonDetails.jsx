import { useState } from 'react';
import PropTypes from 'prop-types';

const PersonDetails = ({
  otherNames,
  ageContext,
  firstName,
  lastName,
  mosaicId,
  dateOfBirth,
  nhsNumber,
  firstLanguage,
  religion,
  dateOfDeath,
  sexualOrientation,
  gender,
  ethnicity,
  addressList,
  phoneNumber,
  email,
  preferredMethodOfContact,
  expandView = false,
}) => {
  const [expandDetails, setExpandDetails] = useState(false);
  const address = addressList?.find(
    ({ displayAddressFlag }) => displayAddressFlag === 'Y'
  );
  return (
    <>
      <div>
        <div className="lbh-table-header">
          <h2 className="govuk-fieldset__legend--m govuk-custom-text-color">
            {expandView ? `${firstName} ${lastName}` : 'PERSON DETAILS'}
          </h2>
          {expandView && (
            <div
              className="govuk-link"
              role="button"
              onClick={() => setExpandDetails(!expandDetails)}
            >
              {expandDetails ? 'Collapse' : 'Expand'} view
            </div>
          )}
        </div>
        <hr className="govuk-divider" />
        {(!expandView || expandDetails) && (
          <>
            <dl className="govuk-summary-list">
              {otherNames?.length > 0 && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Other Names</dt>
                  <dd className="govuk-summary-list__value">
                    <ul className="govuk-list">
                      {otherNames.map(({ firstName, lastName }) => (
                        <li key={otherNames}>
                          {firstName} {lastName}
                        </li>
                      ))}
                    </ul>
                  </dd>
                </div>
              )}
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Person ID</dt>
                <dd className="govuk-summary-list__value">#{mosaicId}</dd>
              </div>
              {gender && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Gender</dt>
                  <dd className="govuk-summary-list__value">{gender}</dd>
                </div>
              )}
              {dateOfBirth && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Date of birth</dt>
                  <dd className="govuk-summary-list__value">
                    {new Date(dateOfBirth).toLocaleDateString('en-GB')}
                  </dd>
                </div>
              )}
              {dateOfDeath && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Date Of Death</dt>
                  <dd className="govuk-summary-list__value">
                    {new Date(dateOfDeath).toLocaleDateString('en-GB')}
                  </dd>
                </div>
              )}
              {ethnicity && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Ethnicity</dt>
                  <dd className="govuk-summary-list__value">{ethnicity}</dd>
                </div>
              )}
              {firstLanguage && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">First Language </dt>
                  <dd className="govuk-summary-list__value">{firstLanguage}</dd>
                </div>
              )}
              {religion && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Religion </dt>
                  <dd className="govuk-summary-list__value">{religion}</dd>
                </div>
              )}
              {sexualOrientation && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">
                    Sexual orientation
                  </dt>
                  <dd className="govuk-summary-list__value">
                    {sexualOrientation}
                  </dd>
                </div>
              )}
              {nhsNumber && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">NHS number</dt>
                  <dd className="govuk-summary-list__value">{nhsNumber}</dd>
                </div>
              )}
              {address?.addressLine1 && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Address</dt>
                  <dd className="govuk-summary-list__value">
                    <p>{address.addressLine1}</p>
                  </dd>
                </div>
              )}
              {phoneNumber?.length > 0 && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Phone Number</dt>
                  <dd className="govuk-summary-list__value">
                    <ul className="govuk-list">
                      {phoneNumber.map(({ phoneNumber, phoneType }) => (
                        <li key={phoneNumber}>
                          {phoneNumber} - {phoneType}
                        </li>
                      ))}
                    </ul>
                  </dd>
                </div>
              )}
              {email && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Email </dt>
                  <dd className="govuk-summary-list__value">{email}</dd>
                </div>
              )}
              {preferredMethodOfContact && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">
                    Preferred Method Of Contact
                  </dt>
                  <dd className="govuk-summary-list__value">
                    {preferredMethodOfContact}
                  </dd>
                </div>
              )}
              {ageContext && (
                <div className="govuk-summary-list__row">
                  <dt className="govuk-summary-list__key">Person type</dt>
                  <dd className="govuk-summary-list__value">
                    {ageContext === 'C' ? 'CFS' : 'ASC'}
                  </dd>
                </div>
              )}
            </dl>
          </>
        )}
      </div>
    </>
  );
};
PersonDetails.propTypes = {
  expandView: PropTypes.bool,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  mosaicId: PropTypes.string,
  dateOfBirth: PropTypes.string,
  nhsNumber: PropTypes.string,
  nationality: PropTypes.string,
  gender: PropTypes.string,
  addressList: PropTypes.arrayOf(PropTypes.shape({})),
  phoneNumber: PropTypes.arrayOf(PropTypes.shape({})),
};
export default PersonDetails;
