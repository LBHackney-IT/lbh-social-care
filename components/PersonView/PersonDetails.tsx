import { useState } from 'react';

import { getEthnicityName } from 'utils/person';
import { Resident } from 'types';

interface Props {
  person: Resident;
  expandView?: boolean;
}

const PersonDetails = ({
  person: {
    otherNames,
    contextFlag,
    firstName,
    lastName,
    id,
    dateOfBirth,
    nhsNumber,
    firstLanguage,
    religion,
    dateOfDeath,
    sexualOrientation,
    gender,
    address,
    ethnicity,
    phoneNumbers,
    emailAddress,
    preferredMethodOfContact,
  },
  expandView = false,
}: Props): React.ReactElement => {
  const [expandDetails, setExpandDetails] = useState(false);
  return (
    <div>
      <div className="lbh-table-header">
        <h2 className="govuk-fieldset__legend--m govuk-custom-text-color">
          {expandView ? `${firstName} ${lastName}` : 'PERSON DETAILS'}
        </h2>
        {expandView && (
          <div
            className="govuk-link"
            role="button"
            onClick={() => setExpandDetails(!expandDetails)}
          >
            {expandDetails ? 'Collapse' : 'Expand'} view
          </div>
        )}
      </div>
      <hr className="govuk-divider" />
      {(!expandView || expandDetails) && (
        <>
          <dl className="govuk-summary-list">
            {otherNames && otherNames?.length > 0 && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Other Names</dt>
                <dd className="govuk-summary-list__value">
                  <ul className="govuk-list">
                    {otherNames.map(({ firstName, lastName }) => (
                      <li key={`${firstName} ${lastName}`}>
                        {firstName} {lastName}
                      </li>
                    ))}
                  </ul>
                </dd>
              </div>
            )}
            <div className="govuk-summary-list__row">
              <dt className="govuk-summary-list__key">Person ID</dt>
              <dd className="govuk-summary-list__value">#{id}</dd>
            </div>
            {gender && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Gender</dt>
                <dd className="govuk-summary-list__value">{gender}</dd>
              </div>
            )}
            {dateOfBirth && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Date of birth</dt>
                <dd className="govuk-summary-list__value">
                  {new Date(dateOfBirth).toLocaleDateString('en-GB')}
                </dd>
              </div>
            )}
            {dateOfDeath && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Date Of Death</dt>
                <dd className="govuk-summary-list__value">
                  {new Date(dateOfDeath).toLocaleDateString('en-GB')}
                </dd>
              </div>
            )}
            {ethnicity && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Ethnicity</dt>
                <dd className="govuk-summary-list__value">
                  {getEthnicityName(ethnicity)}
                </dd>
              </div>
            )}
            {firstLanguage && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">First Language </dt>
                <dd className="govuk-summary-list__value">{firstLanguage}</dd>
              </div>
            )}
            {religion && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Religion </dt>
                <dd className="govuk-summary-list__value">{religion}</dd>
              </div>
            )}
            {sexualOrientation && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Sexual orientation</dt>
                <dd className="govuk-summary-list__value">
                  {sexualOrientation}
                </dd>
              </div>
            )}
            {nhsNumber && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">NHS number</dt>
                <dd className="govuk-summary-list__value">{nhsNumber}</dd>
              </div>
            )}
            {address && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Address</dt>
                <dd className="govuk-summary-list__value">
                  <div>{address.address}</div>
                  <div>{address.postcode}</div>
                </dd>
              </div>
            )}
            {phoneNumbers.length > 0 && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Phone Number</dt>
                <dd className="govuk-summary-list__value">
                  <ul className="govuk-list">
                    {phoneNumbers.map(({ number, type }) => (
                      <li key={number}>
                        {number} - {type}
                      </li>
                    ))}
                  </ul>
                </dd>
              </div>
            )}
            {emailAddress && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Email</dt>
                <dd className="govuk-summary-list__value">{emailAddress}</dd>
              </div>
            )}
            {preferredMethodOfContact && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">
                  Preferred Method Of Contact
                </dt>
                <dd className="govuk-summary-list__value">
                  {preferredMethodOfContact}
                </dd>
              </div>
            )}
            {contextFlag && (
              <div className="govuk-summary-list__row">
                <dt className="govuk-summary-list__key">Person type</dt>
                <dd className="govuk-summary-list__value">
                  {contextFlag === 'C' ? 'CFS' : 'ASC'}
                </dd>
              </div>
            )}
          </dl>
        </>
      )}
    </div>
  );
};

export default PersonDetails;
