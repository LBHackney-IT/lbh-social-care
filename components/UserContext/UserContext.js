import { createContext, useContext, useEffect, useState } from 'react';
import { useRouter } from 'next/router';

import { AUTH_WHITELIST } from 'utils/auth';
import { getUser } from 'utils/api/checkAuth';
import Layout from 'components/Layout';
import Spinner from 'components/Spinner/Spinner';

export const UserContext = createContext({
  user: undefined,
  setUser: () => {},
});

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState();
  const [redirect, setRedirect] = useState();
  const [isLoading, setLoading] = useState(true);
  const router = useRouter();
  useEffect(() => {
    const initializeAuth = async () => {
      try {
        const data = getUser();
        setUser(data);
      } catch (e) {
        setUser();
        setRedirect(e.response.status === 403 ? '/access-denied' : '/login');
      }
      setLoading(false);
    };
    initializeAuth();
  }, []);
  if (isLoading || !router.isReady) {
    return (
      <Layout>
        <Spinner />
      </Layout>
    );
  }
  if (
    typeof window !== 'undefined' &&
    AUTH_WHITELIST.includes(router.pathname) === Boolean(user)
  ) {
    router.push(redirect || location);
    return <></>;
  }
  return (
    <UserContext.Provider
      value={{
        user,
        isLoading,
        setUser,
        redirect,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export function useAuth() {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}
