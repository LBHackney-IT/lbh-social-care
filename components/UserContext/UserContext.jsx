import { createContext, useContext } from 'react';
import { useRouter } from 'next/router';

import { shouldRedirect } from 'utils/auth';
import { isBrowser } from 'utils/ssr';

export const UserContext = createContext({
  user: undefined,
  setUser: () => {},
});

export const AuthProvider = ({ children, user }) => {
  const { pathname, push } = useRouter();
  const redirect = isBrowser() && shouldRedirect(pathname, user);
  if (redirect) {
    push(redirect);
    return <></>;
  }
  return (
    <UserContext.Provider
      value={{
        user,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export function useAuth() {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}
