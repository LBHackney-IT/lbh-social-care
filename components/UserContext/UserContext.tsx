import { createContext, useContext, ReactNode, ReactElement } from 'react';
import { useRouter } from 'next/router';

import { shouldRedirect } from 'utils/auth';
import { isBrowser } from 'utils/ssr';

import type { User } from 'types';

interface Props {
  children: ReactNode;
  user?: User;
}

export const UserContext = createContext<{ user?: User }>({
  user: undefined,
});

export const AuthProvider = ({ children, user }: Props): ReactElement => {
  const { pathname, push } = useRouter();
  const redirect = isBrowser() && shouldRedirect(pathname, user);
  if (redirect) {
    push(redirect);
    return <></>;
  }
  return (
    <UserContext.Provider
      value={{
        user,
      }}
    >
      {children}
    </UserContext.Provider>
  );
};

export const useAuth = (): { user?: User } => {
  const context = useContext(UserContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
};
