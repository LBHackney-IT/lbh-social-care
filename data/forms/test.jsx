const MULTI_SELECT = {
  foo: [
    {
      value: '1',
      text: 'FOO',
    },
    {
      value: '2',
      text: 'Foo',
    },
    {
      value: '3',
      text: 'foo',
    },
  ],
  bar: [
    {
      value: 'a',
      text: 'BAR',
    },
    {
      value: 'b',
      text: 'Bar',
    },
    {
      value: 'c',
      text: 'bar',
    },
  ],
};

export default {
  title: 'Test Form',
  path: '/form/test/',
  successMessage: 'Test Recap',
  steps: [
    {
      id: 'first-step',
      title: 'First Step',
      components: [
        {
          component: 'Radios',
          name: 'show_next_input',
          label: 'Show next input',
          options: [
            { value: 'Y', text: 'Yes' },
            { value: 'N', text: 'No' },
          ],
          rules: { required: true },
        },
        {
          conditionalRender: (data) => data.show_next_input === 'Y',
          component: 'TextInput',
          name: 'conditional_text',
          width: '30',
          label: 'I am the conditional field',
        },
        {
          component: 'Checkbox',
          name: 'show_next_step',
          width: '30',
          label: 'Show next step',
        },
        {
          component: 'Checkbox',
          name: 'show_multi_select_step',
          width: '30',
          label: 'Show multi-select step',
        },
        {
          component: 'Checkbox',
          name: 'show_object_step',
          width: '30',
          label: 'Show object step',
        },
      ],
    },
    {
      id: 'condition-step',
      title: 'Conditional Step',
      conditionalRender: ({ show_next_step }) => show_next_step === true,
      components: [
        {
          component: 'TextInput',
          name: 'title_2',
          width: '30',
          label: 'Title',
        },
      ],
    },
    {
      id: 'object-step',
      title: 'Object Step',
      conditionalRender: ({ show_object_step }) => show_object_step === true,
      components: [
        {
          component: 'ObjectInput',
          name: 'phone_number',
          label: 'Phone Number',
          isInline: true,
          summaryInline: true,
          components: [
            {
              component: 'PhoneInput',
              name: 'phoneNumber',
              label: 'Phone number',
              rules: { required: true },
            },
            {
              component: 'TextInput',
              name: 'phoneType',
              label: 'Phone type',
            },
          ],
        },
      ],
    },
    {
      id: 'multi-select-step',
      title: 'Multi Select Step',
      conditionalRender: ({ show_multi_select_step }) =>
        show_multi_select_step === true,
      components: [
        {
          component: 'Select',
          name: 'first_select',
          label: 'First multi select',
          options: Object.keys(MULTI_SELECT),
        },
        {
          conditionalRender: ({ first_select }) => first_select,
          component: 'Select',
          name: 'multi_select',
          label: 'Second multi select',
          options: ({ first_select } = {}) => MULTI_SELECT[first_select],
        },
      ],
    },
    {
      id: 'last-step',
      title: 'Multi Step',
      isMulti: true,
      components: [
        {
          component: 'TextInput',
          name: 'title_3',
          width: '30',
          label: 'Title',
        },
      ],
    },
  ],
};
