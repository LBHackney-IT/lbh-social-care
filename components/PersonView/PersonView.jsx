import PropTypes from 'prop-types';

import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
import Spinner from 'components/Spinner/Spinner';
import PersonDetails from './PersonDetails';
import { useResident } from 'utils/api/residents';

const PersonView = ({ personId, expandView, children }) => {
  const { data: person, error } = useResident(personId);
  if (error) {
    return <ErrorMessage />;
  }
  if (!person) {
    return <Spinner />;
  }
  return (
    <>
      {!expandView && (
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter govuk-expand-title">
          {person.firstName} {person.lastName}
        </h1>
      )}
      <PersonDetails {...person} expandView={expandView} />
      {typeof children === 'function' ? children(person) : children}
    </>
  );
};

PersonView.propTypes = {
  expandView: PropTypes.bool,
  personId: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export default PersonView;
