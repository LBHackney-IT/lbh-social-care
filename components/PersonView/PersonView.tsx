import Link from 'next/link';

import ErrorMessage from 'components/ErrorMessage/ErrorMessage';
import Spinner from 'components/Spinner/Spinner';
import PersonDetails from './PersonDetails';
import { useResident } from 'utils/api/residents';
import { Resident } from 'types';

interface Props {
  personId: number;
  children?: React.ReactChild | ((arg0: Resident) => React.ReactChild);
  expandView?: boolean;
  showPersonDetails?: boolean;
  canEdit?: boolean;
}

const PersonView = ({
  personId,
  expandView,
  children,
  showPersonDetails = true,
  canEdit,
}: Props): React.ReactElement => {
  const { data: person, error } = useResident(personId);
  if (error) {
    return <ErrorMessage />;
  }
  if (!person) {
    return <Spinner />;
  }
  return (
    <>
      {!expandView && (
        <div className="lbh-table-header">
          <h1 className="govuk-fieldset__legend--l gov-weight-lighter">
            {person.firstName} {person.lastName}
          </h1>
          {canEdit && (
            <Link href={`/people/${person.mosaicId}/update`}>
              <a className="govuk-link">Update person</a>
            </Link>
          )}
        </div>
      )}
      {showPersonDetails && (
        <PersonDetails person={person} expandView={expandView} />
      )}
      {typeof children === 'function' ? children(person) : children}
    </>
  );
};

export default PersonView;
