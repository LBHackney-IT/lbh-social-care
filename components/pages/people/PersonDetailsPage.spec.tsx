import { render, RenderResult, screen } from '@testing-library/react';

import { residentFactory } from '../../../factories/residents';
import { AuthProvider } from '../../UserContext/UserContext';
import PersonView from '../../PersonView/PersonView';
import { userFactory } from '../../../factories/users';
import { createMockedPersonView } from '../../../test/helpers';

import * as casesAPI from 'utils/api/cases';
import { PersonDetailsPage } from './PersonDetailsPage';
import * as warningNotes from '../../../utils/api/warningNotes';
import { mockedWarningNote } from '../../../factories/warningNotes';

jest.mock('../../PersonView/PersonView');
jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: () => ({
    query: { foo: 'bar' },
    replace: jest.fn(),
    pathname: 'foopath',
  }),
}));

const mockedAdultsUser = userFactory.build({
  hasAdminPermissions: false,
  hasUnrestrictedPermissions: false,
  hasChildrenPermissions: false,
  hasAdultPermissions: true,
});

const mockedAdultsResident = residentFactory.build({
  contextFlag: 'A',
});

const mockedChildrensUser = userFactory.build({
  hasAdminPermissions: false,
  hasUnrestrictedPermissions: false,
  hasChildrenPermissions: true,
  hasAdultPermissions: false,
});

const mockedChildrensResident = residentFactory.build({
  contextFlag: 'C',
});

describe('<PersonDetailsPage />', () => {
  beforeEach(() => {
    jest.resetAllMocks();

    jest.spyOn(casesAPI, 'useCasesByResident').mockImplementation(() => ({
      size: 0,
      setSize: jest.fn(),
      data: [],
      isValidating: false,
      mutate: jest.fn(),
      revalidate: jest.fn(),
    }));

    jest.spyOn(warningNotes, 'useWarningNotes').mockImplementation(() => ({
      data: mockedWarningNote,
      mutate: jest.fn(),
      revalidate: jest.fn(),
      isValidating: false,
    }));
  });

  describe('for all users', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedAdultsResident)
      );
      renderResult = render(
        <AuthProvider user={mockedAdultsUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );
    });

    it('should render the details for the person', () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });
  });

  describe('as an adults user', () => {
    it('should show any active warning notes for the person if the person is an adult', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedAdultsResident)
      );

      render(
        <AuthProvider user={mockedAdultsUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText('WARNING NOTE');
    });

    it('should show any active warning notes for the person if the person is a child', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedChildrensResident)
      );

      render(
        <AuthProvider user={mockedAdultsUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText('WARNING NOTE');
    });

    it('should show a restricted banner for a childrens person', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedChildrensResident)
      );

      render(
        <AuthProvider user={mockedAdultsUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText(
        'Some details for this person are restricted due to your permissions.'
      );
    });
  });

  describe('as a childrens user', () => {
    it('should show any active warning notes for the person if the person is a child', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedChildrensResident)
      );

      render(
        <AuthProvider user={mockedChildrensUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText('WARNING NOTE');
    });

    it('should show any active warning notes for the person if the person is an adult', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedAdultsResident)
      );

      render(
        <AuthProvider user={mockedChildrensUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText('WARNING NOTE');
    });

    it('should show a restricted banner for an adult person', () => {
      (PersonView as jest.Mock).mockImplementationOnce(
        createMockedPersonView(mockedAdultsResident)
      );

      render(
        <AuthProvider user={mockedChildrensUser}>
          <PersonDetailsPage personId={100} />
        </AuthProvider>
      );

      screen.getByText(
        'Some details for this person are restricted due to your permissions.'
      );
    });
  });
});
