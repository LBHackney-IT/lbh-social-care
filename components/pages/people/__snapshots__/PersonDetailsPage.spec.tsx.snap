// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonDetailsPage /> for all users should render the details for the person 1`] = `
<DocumentFragment>
  <div
    class="govuk-!-margin-top-7"
  >
    <div
      class=""
    >
      <div
        aria-labelledby="warning-note-title"
        class="govuk-error-summary container"
        role="complementary"
        tabindex="-1"
      >
        <h2
          class="govuk-error-summary__title"
          id="warning-note-title"
        >
          WARNING NOTE
        </h2>
        <div>
          <div
            class="govuk-error-summary__body"
          >
            <div
              class="note"
            >
              <dl
                class="noteDetails"
              >
                <dt>
                  Type
                </dt>
                <dd>
                  Risk to Adults
                </dd>
                <dt>
                  Start date
                </dt>
                <dd>
                  13/11/2020 
                  <i>
                    created by
                  </i>
                   Foo
                </dd>
                <dt>
                  Review date
                </dt>
                <dd>
                  01/12/2020 
                </dd>
              </dl>
              <a
                class="govuk-link govuk-link-underline"
                href="/people/100/warning-notes/1"
              >
                Review / end
              </a>
            </div>
            <div
              class="note"
            >
              <dl
                class="noteDetails"
              >
                <dt>
                  Type
                </dt>
                <dd>
                  Risk to Staff
                </dd>
                <dt>
                  Start date
                </dt>
                <dd>
                  22/11/2020 
                  <i>
                    created by
                  </i>
                   Foo
                </dd>
                <dt>
                  Review date
                </dt>
                <dd>
                  01/01/2021 
                </dd>
              </dl>
              <a
                class="govuk-link govuk-link-underline"
                href="/people/100/warning-notes/2"
              >
                Review / end
              </a>
            </div>
            <p>
              For further information see Warning Note in Records History
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="govuk-!-margin-top-7"
    >
      <div>
        <div
          class="lbh-table-header"
        >
          <h2
            class="govuk-fieldset__legend--m govuk-custom-text-color"
          >
            PERSON DETAILS
          </h2>
        </div>
        <hr
          class="govuk-divider"
        />
        <dl
          class="govuk-summary-list"
        >
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              Person ID
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              #2
            </dd>
          </div>
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              Gender
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              F
            </dd>
          </div>
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              Date of birth
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              13/11/2020
            </dd>
          </div>
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              NHS number
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              12345
            </dd>
          </div>
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              Address
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              <div>
                sjakdjlk
              </div>
              <div>
                hdsadjk
              </div>
            </dd>
          </div>
          <div
            class="govuk-summary-list__row"
          >
            <dt
              class="govuk-summary-list__key"
            >
              Person type
            </dt>
            <dd
              class="govuk-summary-list__value"
            >
              ASC
            </dd>
          </div>
        </dl>
      </div>
    </div>
    <div
      class="govuk-!-margin-top-7"
    >
      <svg
        height="50"
        stroke="#00703c"
        style="display: block; margin: auto;"
        viewBox="0 0 42 42"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(3 3)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              stroke-opacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              transform="rotate(112.708 18 18)"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
    <div
      class="govuk-!-margin-top-7"
    >
      <svg
        height="50"
        stroke="#00703c"
        style="display: block; margin: auto;"
        viewBox="0 0 42 42"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="5"
            transform="translate(3 3)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              stroke-opacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
              transform="rotate(112.708 18 18)"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
    <div
      class="govuk-!-margin-top-7"
    >
      <div>
        <div
          class="lbh-table-header"
        >
          <div>
            <h3
              class="govuk-fieldset__legend--m govuk-custom-text-color"
            >
              RECORDS HISTORY
            </h3>
            <span
              class="govuk-body"
            >
              Linked files are read only
            </span>
          </div>
          <button
            class="lbh-button govuk-button"
            data-module="govuk-button"
          >
            Add a new record
          </button>
        </div>
        <hr
          class="govuk-divider"
        />
        <div>
          <svg
            height="50"
            stroke="#00703c"
            style="display: block; margin: auto;"
            viewBox="0 0 42 42"
            width="50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                stroke-width="5"
                transform="translate(3 3)"
              >
                <circle
                  cx="18"
                  cy="18"
                  r="18"
                  stroke-opacity=".5"
                />
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  transform="rotate(112.708 18 18)"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
